src/doc/unstable-book/src/library-features/asm.md

```diff
@@ -345,6 +345,29 @@ The `h` modifier will emit the register name for the high byte of that register
 
 If you use a smaller data type (e.g. `u16`) with an operand and forget the use template modifiers, the compiler will emit a warning and suggest the correct modifier to use.
 
+## Memory address operands
+
+Sometimes assembly instructions require operands passed via memory addresses/memory locations.
+You could wrap inputs/outputs in `[]` to indicate they as memory operands.
+
+For example:
+
+Old way:
+```rust,allow_fail
+# #![feature(llvm_asm)]
+# fn load_fpu_control_word(control: u16) {
+unsafe { llvm_asm!("fldcw $0" :: "m" (control) :: "volatile") }
+# }
+```
+
+New way:
+```rust,allow_fail
+# #![feature(asm)]
+# fn load_fpu_control_word(control: u16) {
+unsafe { asm!("fldcw [{:e}]", in(reg) control, options(nostack)) }
+# }
```

Comment:
I wonder if specifying the old method as a comment would be better? Also what is `{:e}`, I have never seen that.
```suggestion
You could wrap inputs/outputs in `[]` to indicate they as memory operands. For example:

```rust,allow_fail
# #![feature(asm)]
# fn load_fpu_control_word(control: u16) {
// previously unsafe { llvm_asm!("fldcw $0" :: "m" (control) :: "volatile") }
unsafe { asm!("fldcw [{:e}]", in(reg) control, options(nostack)) }
# }
```

Reply:
> old method as a comment would be better?

The comment is not compiled, also no syntax highlighting.

> And what is `{:e}`? I have never seen that.

You could remove the modifier `:e` and compiler will guide you to use it.
The modifiers are listed here: https://doc.rust-lang.org/nightly/unstable-book/library-features/asm.html#template-modifiers

> I also tried to shorten it as much as possible since the page is already very long.

Thanks, people often read it one time and use searching text to find relevant part they need.
Shorter and more concrete text is nicer, but not a hard constrain, I think.

Reply:
> You could remove the modifier :e and compiler will guide you to use it.

That's because you were previously trying to pass a `u16` into a register instead of a pointer. The `:e` modifier is not needed here since you are passing an address. In fact it is even incorrect for x86_64.

Reply:
Got it. Thanks for correcting me.

Reply:
> The comment is not compiled, also no syntax highlighting.

Yes, it's just a comment. I don't think we need to encourage that.

Reply:
I might say we want the correctness of example, even outdated one.
Also I've never seen the same proposed solution elsewhere.
And the added sections is about 10 lines, not too much.

Reply:
How about moving `For Example` to the previous line?

Reply:
I already did it in the new changes. Or you was talking about something different ?